plugins/armorblox/icon_armorblox/util/api.py

```diff
@@ -0,0 +1,45 @@
+from insightconnect_plugin_runtime.helper import clean
+from insightconnect_plugin_runtime.exceptions import PluginException
+from logging import Logger
+from urllib.parse import urlsplit
+from armorblox.client import Client
+
+
+class ArmorbloxAPI(Client):
+
+    def __init__(self, api_key: str, tenant_name: str, logger = Logger):
+        super().__init__(api_key=api_key, instance_name=tenant_name)
+        self.logger = logger
+        self.incidents_list = []
+
+    def process_incidents(self, params):
+        self.incidents_list = []
+        try:
+            response_json, next_page_token, total_count = self.incidents.list(params=params)
+            self.incidents_list.extend(response_json)
+            while next_page_token:
+                params["page_token"] = next_page_token
+                response_json, next_page_token, total_count = self.incidents.list(params=params)
+                self.incidents_list.extend(response_json)
+        except Exception as credentials_exp:
+            print('Exception has occurred for incorrect credentials. ' + str(credentials_exp))
```

Comment:
I don't think printing is a good practise here. Kindly, please raise `PluginException` instead.